src/pages/audio-rewards-page/ChallengeRewardsTile.tsx

```diff
@@ -55,15 +57,28 @@ const RewardPanel = ({
   openModal,
   progressLabel,
   icon,
-  stepCount
+  stepCount,
+  currentStepCountOverride
 }: RewardPanelProps) => {
   const wm = useWithMobileStyle(styles.mobile)
   const userChallenges = useSelector(getUserChallenges)
 
   const openRewardModal = () => openModal(id)
 
   const challenge = userChallenges[id]
-  const currentStepCount = challenge?.current_step_count || 0
+  const currentStepCount =
+    currentStepCountOverride || challenge?.current_step_count || 0
```

Comment:
I think there's a very subtle bug where if we decrement our step count to zero on the client but discovery is still showing > 0, we prefer discovery still. Realistically probably no impact but would be best to check explicitly against undefined bc of this

Reply:
good catch